### PR TITLE
feat: Add MacAddressRecognizer 

### PR DESCRIPTION
**Pull request by kyoungbinkim**
## Change Description

Adds support for detecting MAC (Media Access Control) addresses as a new global PII entity type. This recognizer supports three industry-standard MAC address formats commonly used across network devices and documentation.

- **Colon-separated**: `00:1A:2B:3C:4D:5E` (IEEE 802 standard)
- **Hyphen-separated**: `00-1A-2B-3C-4D-5E` (Windows format)  
- **Cisco dot-separated**: `0012.3456.789A` (Cisco IOS format)

### Changes Made
- Added `MacAddressRecognizer` in `presidio_analyzer/predefined_recognizers/generic/`
- Pattern-based detection with 0.6 base confidence score
- Context-aware scoring using keywords: `mac`, `mac address`, `hardware address`, `physical address`, `ethernet`
- Validation logic to reject invalid formats and broadcast addresses (`FF:FF:FF:FF:FF:FF`)
- Registered recognizer in `predefined_recognizers/__init__.py`
- Updated `docs/supported_entities.md` with new `MAC_ADDRESS` entity type
- Comprehensive test suite with 15 test cases covering all formats and edge cases

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by kyoungbinkim** [comment]
@RonShakutai  I've resolved all issue. thanks

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    ref : 
+    - https://en.wikipedia.org/wiki/MAC_address#Notational_conventions
+    - https://www.ieee802.org/1/files/public/docs2020/yangsters-smansfield-mac-address-format-0420-v01.pdf
+    """
+
+    PATTERNS = [
+        Pattern(
+            "MAC_ADDRESS",
+            r"\b([0-9A-Fa-f]{2}[:\-]){5}[0-9A-Fa-f]{2}\b",
```
Comment (RonShakutai) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py]: Do we want to detect only “pure colon” OR “pure hyphen” MACs, and reject mixed ones like 00:1A-2B:3C-4D:5E?

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - Cisco format (dot-separated groups of 4): 0012.3456.789A
+
+    ref : 
+    - https://en.wikipedia.org/wiki/MAC_address#Notational_conventions
+    - https://www.ieee802.org/1/files/public/docs2020/yangsters-smansfield-mac-address-format-0420-v01.pdf
+    """
+
+    PATTERNS = [
+        Pattern(
+            "MAC_ADDRESS",
```
Comment (RonShakutai) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py]: should we use more descrpitive names for the pattern ? 
i have thought about something like this, maybe in capital letters... 
up to you

```python
Pattern("mac_colon_or_hyphen", r"...", 0.6),
Pattern("mac_cisco_dot", r"...", 0.6),

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        """
+        # Remove separators and validate hex characters and length
+        cleaned = re.sub(r'[:\-.]', '', pattern_text)
+
+        # MAC address must be exactly 12 hex characters
+        if len(cleaned) != 12:
+            return True
+
+        # All characters must be valid hex
+        if not all(c in '0123456789ABCDEFabcdef' for c in cleaned):
```
Comment (RonShakutai) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py]: Should we use this re.fullmatch and skip the for ? 
  
if re.fullmatch(r"[0-9A-Fa-f]{12}", cleaned) is None:
    return True

File: presidio-analyzer/tests/test_mac_recognizer.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import pytest
+
```
Comment (RonShakutai) [inline presidio-analyzer/tests/test_mac_recognizer.py:2]: should we add a bit more cases like 
Lowercase MAC, Uppercase MAC, Mixed case MAC, Null address 00:00:00:00:00:00

**Review by RonShakutai — APPROVED** [review]
Great PR, 
Left a minor comments

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py (hunk 1 above)
Comment (kyoungbinkim) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py]: You're right. That's why I modified the regular expression. Thanks.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py (hunk 2 above)
Comment (kyoungbinkim) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py]: Thanks for the suggestion. I changed the name to be more specific based on your suggestion. Thank you.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py (hunk 3 above)
Comment (kyoungbinkim) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/generic/mac_recognizer.py]: I think your way is better. I modified it.

File: presidio-analyzer/tests/test_mac_recognizer.py (hunk 4 above)
Comment (kyoungbinkim) [reply presidio-analyzer/tests/test_mac_recognizer.py:2]: I added several test cases.

**Comment by kyoungbinkim** [comment]
@RonShakutai  Thanks so much for the code review. I've made the changes you suggested.

**Comment by RonShakutai** [comment]
In my opinion, this recognizer should be included in the defaults, WDYT? @omri374  @SharonHart

**Comment by RonShakutai** [comment]
> @RonShakutai Thanks so much for the code review. I've made the changes you suggested.

Great, could you review the failing steps in the CI and update the PR as needed?

**Comment by kyoungbinkim** [comment]
@RonShakutai  I've resolve lint error. thanks

File: presidio-analyzer/presidio_analyzer/conf/default_recognizers.yaml [hunk 5]
```diff
@@ -192,6 +192,9 @@ recognizers:
   - name: MedicalLicenseRecognizer
     type: predefined
 
+  - name: MacAddressRecognizr
```
Comment (SharonHart) [inline presidio-analyzer/presidio_analyzer/conf/default_recognizers.yaml]: typo

Comment (kyoungbinkim) [reply presidio-analyzer/presidio_analyzer/conf/default_recognizers.yaml]: I fixed this typo. thanks

**Comment by RonShakutai** [comment]
Please include the name parameter as done in the other recognizers. @kyoungbinkim 
https://github.com/microsoft/presidio/actions/runs/21025612517/job/60451748508?pr=1829

**Comment by kyoungbinkim** [comment]
@RonShakutai  Sorry for the inconvenience. I just fixed it by adding the name parameter to the constructor. Thanks.

**Comment by kyoungbinkim** [comment]
https://github.com/kyoungbinkim/presidio/actions/runs/21051121722/job/60537402808
